### ISSUE TITLE
Hide MB_EMBEDDING_SECRET_KEY from embedding code when set as an environment variable

### ISSUE THREAD

**Issue by Tony-metabase** [metabase/metabase#30354]
**Is your feature request related to a problem? Please describe.**

The MB_EMBEDDING_SECRET_KEY can be considered as sensitive information and shouldn't be exposed that easily, we also mention this on our documentation:

![image](https://user-images.githubusercontent.com/110378427/234227927-5eaea0d1-aeb0-4123-84ad-12496d3ac1a5.png)

In the admin panel when the key is set through an environment variable we don't expose this and simply show

![image](https://user-images.githubusercontent.com/110378427/234228277-e699a94f-8f22-4965-8f8d-994f5ec3023e.png)

On the other hand when you proceed to embed the dashboard the small code snipped ends up exposing this Secret Key

<img width="1379" alt="image" src="https://user-images.githubusercontent.com/110378427/234228954-98626dc7-2bb6-4a3c-9b35-734a560070f8.png">

**Describe the solution you'd like**
Make the two paths consistent ... So if in the Admin tab is not exposed because it has been set by an Environment Variable then the Embedding code snipped should show a similar output. Something like

`var METABASE_SECRET_KEY = "This has been set by the MB_EMBEDDING_SECRET_KEY environment variable."`

**Describe alternatives you've considered**
None

**Comment by paoliniluis** [comment]
It will only be available for admins, but the issue makes absolute sense

**Comment by filipesilva** [comment]
I'm not sure I follow; the proposed solution would make the code snippet unusable, right? The var would not contain the secret, and thus when the code runs auth would fail.

You could show an invalid snippet, and then copy the correct code to the clipboard when clicking the copy button. The user would still get incorrect code if they copied the text without the copy button though, which I think isn't very user friendly. 

You could also take extra action to prevent copying without the button, like preventing text selection, copy shortcuts, or right click menu. But this feels like solving problems introduced by the proposed solution, rather than solving user problems.

**Comment by Tony-metabase** [comment]
Yes that would be a response of hiding the MB_EMBEDDING_SECRET_KEY from the code ... Agree that adding an extra step is not ideal but there isn't much one can do to not expose the SECRET_KEY or at least I couldn't think of another one.

If we are saying in our documentation that it should be treated like a password then we should also expose it as such ... METABASE_SECRET_KEY = *******************

**Comment by filipesilva** [comment]
Also worth keeping in mind what happens after copying the secret. Now the user has the snippet on the clipboard, and that snippet contains the real secret.

Presumably they will paste it somewhere, thus making it show on screen.

**Comment by albertoperdomo** [comment]
My bias is to:
* not consider this a bug, but a feature request
* increase the scope to make it about not exposing the secret unless necessary, regardless of how it was set

**Comment by dragonsahead** [comment]
I think we fixed this recently right @Tony-metabase ? please close it if so

**Comment by Tony-metabase** [comment]
We did not but Guest Embed doesn't expose this preview section so good to go ... Just move away from static embedding
